### PR TITLE
Add support for newrelic-log-ingestion-xxxx name

### PR DESCRIPTION
**Pull request by chaudharysaket**
Add support for newrelic-log-ingestion-xxxx name
* Install using new `newrelic-log-ingestion-<stackid>` name.
* Maintain backwards compatibility with existing installations by using old newrelic-log-ingestion name.
* This fixes https://github.com/newrelic/newrelic-lambda-cli/issues/240.

### CODE REVIEW

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:571]: Since the log ingestion function will only exist if the stack exists we can just check the stack status and remove this function check. This was here before because previously we could only install one log ingestion function per region but now we can install multiple per region so this isn't needed. If the old log ingestion function exists this should be fine as it is now unused though we may want to notify the user that they should remove it. I've made some suggested edits that remove this function if-else check but after that you'll have to tab in the remaining inner conditional block that checks the stack status.

Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:571]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:572]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:595]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:596]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:597]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:598]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:599]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:600]:
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:587]: Let's change the failure message here to match the new behavior:
```suggestion
        else:
            failure(
                "CloudFormation Stack NewRelicLogIngestion exists (status: %s).\n"
                "Please manually delete the stack and re-run this command."
                % stack_status
            )
            return False
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:570]: Maybe add something here warning users that the old log ingestion function is present in their region and they should delete it:
```suggestion
    assert isinstance(input, IntegrationInstall)
    function = get_function(input.session, "newrelic-log-ingestion")
    if function:
        warning("It looks like an old log ingestion function is present in this region. Consider manually deleting this as it is no longer used and has been replaced by a log ingestion function specific to the stack.")
```

File: newrelic_lambda_cli/subscriptions.py
Comment (hmstepanek) [inline newrelic_lambda_cli/subscriptions.py:87]: Again here, we might consider adding a message about old log ingestion functions.
```suggestion
    assert isinstance(input, SubscriptionInstall)
    function = get_function(input.session, "newrelic-log-ingestion")
    if function:
        warning("It looks like an old log ingestion function is present in this region. Consider manually deleting this as it is no longer used and has been replaced by a log ingestion function specific to the stack.")
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:92]: Just return None (which is the default behavior if you don't specify a return at all) if the stack does not exist.
```suggestion
```

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:395]: This looks like it's grabbing settings from the existing log ingestion function, if the existing log ingestion function is the old function name we should grab them from the old name here and again warn the user about having the old log ingestion function in their region. Something like the following:
```suggestion
        unique_log_ingestion_name = get_unique_newrelic_log_ingestion_name(input.session)
        old_props = {}
        try:
            old_props = lambda_client.get_function_configuration(
                FunctionName=unique_log_ingestion_name
            )
        except Lambda.Client.exceptions.ResourceNotFoundException:
            old_props = lambda_client.get_function_configuration(
                FunctionName="newrelic-log-ingestion"
            )
            warning("It looks like an old log ingestion function is present in this region. Pulling the settings from that. Consider manually deleting this as it is no longer used and has been replaced by a log ingestion function specific to the stack.")
```

**Review by hmstepanek — CHANGES_REQUESTED** [review]
I think this looks pretty good. I requested some small changes-mainly warning the user about having the old function name in their region.

Have you manually verified this against a real aws lambda? I'd try installing one from scratch and also install one using the old method and trying to update it with the new method.

File: newrelic_lambda_cli/integrations.py
Comment (chaudharysaket) [reply newrelic_lambda_cli/integrations.py:93]: The function needs to return string for the nested stack in the update_log_ingestion_function which in turn calls [import_log_ingestion_function](https://github.com/newrelic/newrelic-lambda-cli/pull/254/files#:~:text=Starting%20import%22)

File: newrelic_lambda_cli/integrations.py
Comment (chaudharysaket) [reply newrelic_lambda_cli/integrations.py:587]: addressed in the revised PR

File: newrelic_lambda_cli/integrations.py
Comment (chaudharysaket) [reply newrelic_lambda_cli/integrations.py]: Yes I have tested by installing the CLI locally and using it for integrating the lambda and adding subscription.

File: newrelic_lambda_cli/integrations.py
Comment (hmstepanek) [inline newrelic_lambda_cli/integrations.py:403]: We probably need to pass this name (unique_log_ingestion_name) into the `_import_log_ingestion_function` since here the stack id exists and by the time it calls `_import_log_ingestion_function` it doesn't.

**Review by hmstepanek — CHANGES_REQUESTED** [review]
This is looking really good! We added the stack name to the subscriptions but do you think we should add it to the integrations as well? To your point you made earlier in slack, it seems like it doesn't truly allow multiple log ingestion functions per region until it allows multiple stacks per region. Seems like we shouldn't be hardcoding the stack name at all anymore.

File: newrelic_lambda_cli/cli/subscriptions.py
Comment (chaudharysaket) [reply newrelic_lambda_cli/cli/subscriptions.py:43]: Addressed in the revision

**Review by hmstepanek — APPROVED** [review]
Looks good! Please do a rebase on this branch with the latest version of main. Our team recently added Python3.12 support while this was in PR. Along with that change we also changed the latest version to 0.8.0 which means in this PR we need to bump the version to 0.9.0 now. After those changes are made please feel free to merge this. I'm going to mark this PR as pre-approved assuming you make those changes. Once this is merged we'll need to publish a new release by creating a new github release with release notes (I believe you can auto-generate these) when creating the github release.

**Review by hmstepanek — APPROVED** [review]

**Pull request merged**
